common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -2547,6 +2547,9 @@ public static enum ConfVars {
         "If this config is true only pushed down filters remain in the operator tree, \n" +
         "and the original filter is removed. If this config is false, the original filter \n" +
         "is also left in the operator tree at the original place."),
+    HIVE_JOIN_DISJ_TRANSITIVE_PREDICATES_PUSHDOWN("hive.optimize.join.disjunctive.transitive.predicates.pushdown",
+        true, "Whether to transitively infer disjunctive predicates across joins. \n"
+            + "Disjunctive predicates can lead to OOM in transitive inference."),
```

Comment:
Suggestion: Disjunctive predicates are hard to simplify and pushing them down may in some cases lead to infinite rule matching causing stackoverflow and OOM errors.